control-plane/subcommand/common/common_test.go

```diff
@@ -199,15 +299,16 @@ func TestGetResolvedServerAddresses(t *testing.T) {
 // startMockServer starts an httptest server used to mock a Consul server's
 // /v1/acl/login endpoint. apiCallCounter will be incremented on each call to /v1/acl/login.
 // It returns a consul client pointing at the server.
-func startMockServer(t *testing.T, apiCallCounter *int) (*api.Client, *api.Config) {
-
+func startMockServer(t *testing.T) *api.Client {
```

Comment:
we don't need the counter anymore I think. Previously, the mock server always returned `testLoginResponse` regardless of the URL path and method, and so we instead relied on the counter to make sure this call is made. If we only return successful response for the right path+method, we don't need the counter.